[Bug] Material not showing after upgrade from Iron to Jazzy

## Description

We currently try to upgrade our stack from iron to jazzy. We mostly use stl files in our model. With rviz in iron all meshes + materials are shown correctly. In jazzy all meshes based on a stl file are shown in the same light gray (see screenshot)

![image](https://github.com/ros2/rviz/assets/8481806/21d55f69-1d4b-4aa4-a419-3fd9c731e1a1)


Additionally rviz produces (Only in case the model is loaded via the moveit planning scene) and endless stream of such errors:

> [rviz2-17] [ERROR] [1719212675.170806883] [rviz2]: Can't assign material Shape86Material to the ManualObject MeshShape_ManualObject7 because this Material does not exist in group General. Have you forgotten to define it in a .material script?
[rviz2-17] [ERROR] [1719212675.171172805] [rviz2]: Can't assign material Shape88Material to the ManualObject MeshShape_ManualObject8 because this Material does not exist in group General. Have you forgotten to define it in a .material script?
[rviz2-17] [ERROR] [1719212675.171424283] [rviz2]: Can't assign material Shape90Material to the ManualObject MeshShape_ManualObject9 because this Material does not exist in group General. Have you forgotten to define it in a .material script?
[rviz2-17] [ERROR] [1719212675.171593158] [rviz2]: Can't assign material Shape92Material to the ManualObject MeshShape_ManualObject10 because this Material does not exist in group General. Have you forgotten to define it in a .material script?

It is actually this stream of endless errors that makes it unusable. 

## System

- jazzy binary install
- rviz version: 14.1.1

## Additional information

For most of these meshes a material is set in the `<visual>` block such as `<material name="steel_blue"/>`

As previously mentioned in the thread of issue #1236 I keep getting endless rviz2 logging errors even after the latest update to version 14.1.3 of rviz2 which included a fix for issue #1236, but not for me. In my case the objects producing the continuous rviz2 log errors are collision objects that were added using moveit2's applyCollisionObject methods and they include .stl meshes (note: the objects are not defined in a URDF file). 

The typical error message is the following, with the numbers between "Shape" and "Material" as well as after "Object" being kind of an computed counter increasing over time:

`[rviz2-3] [ERROR] [1721749218.192680538] [rviz2]: Can't assign material Shape18906Material to the ManualObject MeshShape_ManualObject17738 because this Material does not exist in group General. Have you forgotten to define it in a .material script?`

I am guessing here, but it seems that it kind of _generates_ materials (like "Shape18906Material" from log output cited above) counting upwards the whole time and it looks for this material in some group "General" which I am not familiar with. 


My current setup is

- Ubuntu 24.04
- ROS2 Jazzy Debian Packages installed via apt
- Currently installed rviz2 version: `ros-jazzy-rviz2/noble,now 14.1.3-1noble.20240719.202006 amd64 [installed,automatic]`

Any ideas?

--- a/rviz_rendering/src/rviz_rendering/objects/mesh_shape.cpp
+++ b/rviz_rendering/src/rviz_rendering/objects/mesh_shape.cpp
@@ -74,7 +74,8 @@ void MeshShape::beginTriangles()
 
   if (!started_) {
     started_ = true;
-    manual_object_->begin(material_name_, Ogre::RenderOperation::OT_TRIANGLE_LIST);
+    manual_object_->begin(material_name_, Ogre::RenderOperation::OT_TRIANGLE_LIST,
+      "rviz_rendering");
   }
 }
 

